extensions/replication/utils/ObjectFailureEntry.js

```diff
@@ -34,8 +37,17 @@ class ObjectFailureEntry {
         return this.sitename;
     }
 
+    getReplicationRoles() {
+        return this.role;
+    }
+
     getMember() {
-        return `${this.bucket}:${this.objectKey}:${this.encodedVersionId}`;
+        const member =
+            `${this.bucket}:${this.objectKey}:${this.encodedVersionId}`;
+        if (this.role) {
+            return `${member}:${this.role}`;
+        }
+        return member;
```

Comment:
```suggestion
        if (this.role) {
            return `${member}:${this.role}`;
        }
        return `${this.bucket}:${this.objectKey}:${this.encodedVersionId}`;
```

Reply:
`member` is `${this.bucket}:${this.objectKey}:${this.encodedVersionId}` which is reused when specifying a role.